src/optable.ts

```diff
@@ -225,7 +225,21 @@ optable[opcodes.PRINT_NEWLINE] = function(f: Py_FrameObject) {
 }
 
 optable[opcodes.RETURN_VALUE] = function(f: Py_FrameObject, t: Thread) {
-    t.asyncReturn(f.pop());
+    // If a generator frame is exhausted, we raise StopIteration exception
+    // StopIteration exceptions generated by generators inside a for loop are ALWAYS caught
+    // So we set up an EXCEPT_HANDLER, AFTER a generator has been exhausted, and then raise StopIteration which is immediately caught
+    if (f.genFrame) {
+        f.genFrame = false;
+
+        // If generator was called inside a loop,  
+        if (t.loop_exc_block.length > 0) {
+            (<Py_FrameObject>f.back).blockStack.push(t.loop_exc_block.pop());
```

Comment:
Is it OK that you're reversing the order of these?

(e.g. if `loop_exc_block` contains values [1,2,3], you're pushing it onto `blockStack` as [3,2,1])